server/application.js

```diff
@@ -1,55 +1,77 @@
 
+// app.use(morgan('dev', {
+//   skip: (req, res) => res.statusCode < 400,
+//   stream: process.stderr,
+// }));
+
+// app.use(morgan('dev', {
+//   skip: (req, res) => res.statusCode >= 400,
+//   stream: process.stdout,
+// }));
+
 app.use('/restaurant/:restaurantId', express.static(path.join(__dirname, '../public/index.html')));
 app.use('/menusBundle.js', express.static(path.join(__dirname, '../public/dist/bundle.js')));
 
 app.get('/menus/restaurant/:restaurantId/menu', (req, res) => {
-  db.retrieve(req.params.restaurantId, (err, results) => {
-    if (err && err.message.includes('Cast to number failed for value "NaN"')) {
-      res.status(400).json('Bad request');
-    } else if (err) {
+  const text = `
+  SELECT r.rest_name, r.rest_headers, i.item_id, i.item_description, i.item_price, i.item_url, s.section_name, d.restriction_name
+  FROM restaurants r
+  INNER JOIN restaurant_items ri on r.rest_id = ri.rest_id
+  INNER JOIN menu_items i on ri.item_id = i.item_id
+  INNER JOIN item_section si on i.item_id = si.item_id
+  INNER JOIN menu_sections s on si.section_id = s.section_id
+  LEFT JOIN item_restrictions ir on i.item_id = ir.item_id
+  LEFT JOIN dietary_restrictions d on ir.restriction_id = d.restriction_id WHERE r.rest_id = $1;`;
+  const values = [req.params.restaurantId];
+  db.query(text, values, (err, result) => {
+    if (err) {
       res.status(500).json('Unable to retrieve menu data from database');
     } else {
-      res.status(200).json(results);
+      res.status(200).json(result);
     }
   });
 });
 
 app.post('/menus/restaurant/:restaurantId/menu', (req, res) => {
-  db.post(req.params.restaurantId, req.body, (err, results) => {
+  const text = `
+  INSERT INTO menu_items (item_name, item_description, item_price, item_url)
+  VALUES ('Blueberry Waffles', 'v delicious', 7.50, 'https://s3-us-west-1.amazonaws.com/sdc-datatable/menu-item1.jpg')`;
+  db.query(text, (err, result) => {
     if (err) {
-      res.sendStatus(500);
-      throw err;
+      res.status(500).json('Unable to insert item into menu');
     } else {
       res.status(201).json(result);
     }
-  })
+  });
 });
 
 app.put('/menus/restaurant/:restaurantId/menu', (req, res) => {
-  db.put(req.params.restaurantId, req.body, (err, results) => {
+  const text = 'UPDATE menu_items SET item_price = 8.50 WHERE item_id=1050;';
+  db.query(text, (err, result) => {
     if (err) {
       res.sendStatus(500);
       throw err;
     } else {
       res.status(204).json(result);
     }
-  })
+  });
 });
 
 app.delete('/menus/restaurant/:restaurantId/menu', (req, res) => {
-  db.delete(req.params.restaurantId, req.body, (err, results) => {
+  const text = 'DELETE FROM menu_items WHERE item_id=1050;';
+  db.query(text, (err, result) => {
     if (err) {
-      res.sendStatus(500);
-      throw err;
+      res.status(500).json('Unable to retrieve menu data from database');
     } else {
       res.status(200).json(result);
     }
-  })
+  });
 });
 
```

Comment:
Looks like the only method here that's accepting values from the URL is the GET request, are these POST/PUT/DELETE functions all temporary? Sadly, if so, I don't think it'll be a useful exercise for stress testing your system because it is just going to always re-issue the exact same SQL each time the VERB/URL is hit, which means that your database will very quickly recognize that query and cache that single response on its own which somewhat defeats the purpose of the test where we're aiming to simulate more real-life data.

Lemee know whatcha think.

Reply:
Agreed, I left it simple since I was having trouble with transactions and didn't want to fall behind but I went back and further built them out (got transactions to work!)

Reply:
👍